### PR TITLE
[iOS] System keyboard + Reachability crash fix

### LINKED ISSUE #1166: [iOS] crash attempting to call SCNetworkReachabilityScheduleWithRunLoop
The following crash is happening quite frequently (iOS 10.3, 11.1, 11.2, 11.3 and 11.4 on iPhones and iPads) when attempting to set up a notification/callback to alert Keyman to changes in reachability:

```
Thread 0#0	(null) in __semwait_signal_nocancel ()
#1	(null) in nanosleep$NOCANCEL ()
#2	(null) in sleep$NOCANCEL ()
#3	(null) in ConnectToServer ()
#4	(null) in DNSServiceCreateConnection ()
#5	(null) in nw_resolver_instantiate_dns_connection_for_parameters ()
#6	(null) in nw_resolver_create_dns_service_locked ()
#7	(null) in nw_resolver_set_update_handler ()
#8	(null) in __SCNetworkReachabilityRestartResolver ()
#9	(null) in __SCNetworkReachabilitySetDispatchQueue ()
#10	(null) in SCNetworkReachabilityScheduleWithRunLoop ()
#11	0x0000000100411280 in -[Reachability startNotifier] at /Users/tom/Documents/keymanapp/keyman/ios/engine/KMEI/KeymanEngine/lib/Reachability/Reachability.m:137
#12	0x000000010043df74 in Manager.init() at /Users/tom/Documents/keymanapp/keyman/ios/engine/KMEI/KeymanEngine/Classes/Manager.swift:186
#13	(null) in @objc Manager.init() ()
#14	(null) in globalinit_33_07B12AF05789801466470386CC680AA0_func13 ()
#15	(null) in _dispatch_client_callout ()
#16	(null) in dispatch_once_f$VARIANT$mp ()
#17	0x000000010049c918 in InputViewController.init(nibName:bundle:) at /Users/tom/Documents/keymanapp/keyman/ios/engine/KMEI/KeymanEngine/Classes/Manager.swift:58
#18	(null) in @objc InputViewController.init(nibName:bundle:) ()
#19	0x000000010030ce30 in KeyboardViewController.init(nibName:bundle:) at /Users/tom/Documents/keymanapp/keyman/ios/engine/KMEI/SystemKeyboard/KeyboardViewController.swift:22
#20	(null) in @objc KeyboardViewController.init(nibName:bundle:) ()
#21	(null) in -[_UIViewServiceViewControllerOperator __createViewController:withContextToken:fbsDisplays:appearanceSerializedRepresentations:legacyAppearance:traitCollection:initialInterfaceOrientation:hostAccessibilityServerPort:replyHandler:] ()
#22	(null) in __invoking___ ()
#23	(null) in -[NSInvocation invoke] ()
#24	(null) in -[NSInvocation invokeWithTarget:] ()
#25	(null) in -[_UIViewServiceImplicitAnimationDecodingProxy forwardInvocation:] ()
#26	(null) in ___forwarding___ ()
#27	(null) in _CF_forwarding_prep_0 ()
#28	(null) in __invoking___ ()
#29	(null) in -[NSInvocation invoke] ()
#30	(null) in -[NSInvocation invokeWithTarget:] ()
#31	(null) in -[_UIQueueingProxy forwardInvocation:] ()
#32	(null) in ___forwarding___ ()
#33	(null) in _CF_forwarding_prep_0 ()
#34	(null) in __invoking___ ()
#35	(null) in -[NSInvocation invoke] ()
#36	(null) in -[NSInvocation invokeWithTarget:] ()
#37	(null) in ___forwarding___ ()
#38	(null) in _CF_forwarding_prep_0 ()
#39	(null) in __invoking___ ()
#40	(null) in -[NSInvocation invoke] ()
#41	(null) in _dispatch_client_callout ()
#42	(null) in _dispatch_block_invoke_direct$VARIANT$mp ()
#43	(null) in __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__ ()
#44	(null) in -[FBSSerialQueue _performNext] ()
#45	(null) in -[FBSSerialQueue _performNextFromRunLoopSource] ()
#46	(null) in __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ ()
#47	(null) in __CFRunLoopDoSources0 ()
#48	(null) in __CFRunLoopRun ()
#49	(null) in CFRunLoopRunSpecific ()
#50	(null) in GSEventRunModal ()
#51	(null) in UIApplicationMain ()
#52	(null) in _xpc_objc_main ()
#53	(null) in xpc_main ()
#54	(null) in -[NSXPCListener resume] ()
#55	(null) in -[PKService run] ()
#56	(null) in +[PKService main] ()
#57	(null) in +[PKService _defaultRun:arguments:] ()
#58	(null) in NSExtensionMain ()
#59	(null) in start ()
```

It appears to be this exact same error in a very similar scenario to ours:
https://stackoverflow.com/questions/40225201/firebase-app-configuration-crash
Unfortunately, no solution has been given. There are multiple versions of the "reachability sample" code floating around on the web, but this particular bit of code where the error is happening seems to be pretty consistent, so it might indeed be some kind of Firebase/keyboard extension incompatibility.

Since the reachability code doesn't seem to be absolutely essential, I think the solution is perhaps to try to wrap the offending call in a do/catch block.

### PR DESCRIPTION
**Pull request by jahorton**
Fixes #1166.

Due to some unfortunate mis-organization of code, we had internet "reachability" checks running from the system keyboard.  This makes sense in the app, which can check for keyboard updates and has event handlers designed toward that end, but little sense for the system keyboard - and since the handlers weren't intended for the system keyboard, the latter has been causing crashes.

Easiest solution?  Just disable reachability checks for system-keyboard contexts.

Also, Xcode 10's been causing some issues with IDE-based builds that are resolved by using the legacy/"original" build system.  Since it's pretty helpful to use the Simulator for testing through Xcode, the workspace settings change is being included.

### CODE REVIEW

File: ios/history.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 * Bookmark add button is enabled only when title/url fields have text (#1073)
 * Removed deprecated code to to support keyman:// scheme for ad-hoc distribution (#1160)
 * Updated the default keyboard to SIL EuroLatin (#1101)
+* Fixed bug behind some crashes of system keyboard (#1166)
```
Comment (darcywong00) [inline ios/history.md]: FYI, while you were gone, @mcdurdin went with the direction of history.md files referring to PR numbers (in the past, we used issue number)

File: ios/engine/KMEI/KeymanEngine/Classes/Manager.swift [hunk 2]
```diff
@@ -181,9 +181,12 @@ public class Manager: NSObject, HTTPDownloadDelegate, UIGestureRecognizerDelegat
     keymanWeb.view.addGestureRecognizer(hold)
 
     reachability = Reachability(hostName: keymanHostName)
```
Comment (darcywong00) [inline ios/engine/KMEI/KeymanEngine/Classes/Manager.swift]: Should `reachability` only be initialized for non-system keyboards?

Comment (jahorton) [reply ios/engine/KMEI/KeymanEngine/Classes/Manager.swift]: I'm pretty sure keyboard updates are only done from within the app, and never from the system keyboard.  Is it the same for the Android app?

Comment (darcywong00) [reply ios/engine/KMEI/KeymanEngine/Classes/Manager.swift]: For the Android app, keyboard updates can be downloaded from the system keyboard.
(I tested with an app the uses european2-1.0.js as the default keyboard)

**Comment by mcdurdin** [comment]
Long term we need to move updates into a background async process that uses
system notifications or similar to let user accept them with minimum of
interference.

**Comment by jahorton** [comment]
@mcdurdin Shall I punt on that for now and make it a separate issue for a later release cycle?

**Comment by darcywong00** [comment]
> and make it a separate issue for a later release cycle?

I already created issues 1316 and 1317 (see links above)

**Pull request merged**